network/endpoint_test.go

```diff
@@ -374,4 +376,68 @@ var _ = Describe("Test Endpoint", func() {
 			})
 		})
 	})
+
+	// validation when calling add
+	Describe("Test validateEndpoints", func() {
+		Context("When in a single add call we create two endpoint structs", func() {
+			It("Should have the same container id", func() {
+				eps := []*endpoint{
+					{
+						ContainerID: "0ea7476f26d192f067abdc8b3df43ce3cdbe324386e1c010cb48de87eefef480",
+						NICType:     cns.InfraNIC,
+					},
+					{
+						ContainerID: "0ea7476f26d192f067abdc8b3df43ce3cdbe324386e1c010cb48de87eefef480",
+						NICType:     cns.DelegatedVMNIC,
+					},
+				}
+				Expect(validateEndpoints(eps)).To(BeNil())
+			})
+		})
+		Context("When in a single add call we have different container ids", func() {
+			It("Should error", func() {
+				eps := []*endpoint{
+					{
+						ContainerID: "0ea7476f26d192f067abdc8b3df43ce3cdbe324386e1c010cb48de87eefef480",
+						NICType:     cns.InfraNIC,
+					},
+					{
+						ContainerID: "0ea7476f26d192f067abdc8b3df43ce3cdbe324386e1c010cb48de87eefef481",
+						NICType:     cns.DelegatedVMNIC,
+					},
+				}
+				Expect(validateEndpoints(eps)).ToNot(BeNil())
+			})
+		})
+		Context("When no container id", func() {
```

Comment:
let's also add a case when no container id for ib NIC because we found that issue before for both happy and unhappy path